CIP-0062/README.md

```diff
@@ -0,0 +1,473 @@
+
+Errors: [`APIError`](#extended-apierror)
+
+The `cardano.{walletName}.governance.enable()` method is used to enable the
+governance API. It should request permission from the wallet to enable the API for the requested purposes.
+
+If permission is granted, the rest of the API will be available. The wallet
+should maintain a specific whitelist of allowed clients and voting purposes for
+this API. This whitelist can be used to avoid asking for permission every time.
+
+This api, being an extension of
+[CIP-30 (Cardano dApp-Wallet Web Bridge)](https://cips.cardano.org/cips/cip30/),
+expects that `cardano.{walletName}.enable()` to be enabled and added to CIP-30
+whitelist implicitly.
+
+When both this API and [CIP-30](https://cips.cardano.org/cips/cip30/) being
+enabled, is up to the wallet to decide the number of prompts requesting
+permissions to be displayed to the user.
+
+* `purpose` - this is a list of purposes that the dApp is advising that it will be using on the API.  The wallet must respond with an error if the purpose is not supported by the Wallet.
+
+  Note: Currently only voting purpose 0 (Catalyst) is defined. The wallet should reject any other purpose requested.
+
+### Returns
+
+Upon successful connection via
+[`cardano.{walletName}.governance.enable()`](#cardanowalletnamegovernanceenablepurpose-votingpurpose-promiseapi),
+a javascript object we will refer to as `API` (type) / `api` (instance) is
+returned to the dApp with the following methods.
+
+## Governance API
+
+Except `signVotes`, no other method should require any user
+interaction as the user has already consented to the dApp reading information
+about the wallet's governance state when they agreed to
+[`cardano.{walletName}.governance.enable()`](#cardanowalletnamegovernanceenablepurpose-votingpurpose-promiseapi).
+The remaining methods
+[`api.signVotes()`](#apisignvotesvotes-vote-promisebytes) and
+[`api.signData()`](#apisubmitdelegationdelegation-delegation-promisesigneddelegationmetadata)
+must request the user's consent in an informative way for each and every API
+call in order to maintain security.
+
+The API chosen here is for the minimum API necessary for dApp <-> Wallet
+interactions without convenience functions that don't strictly need the wallet's
+state to work.
+
+## api.signVotes(votes: Vote[], settings: string): Promise\<Bytes>[]
+
+Errors: [`APIError`](#extended-apierror), [`TxSignError`](#extended-txsignerror)
+
+* `votes` - an array of up to 10 votes to be validated with the wallet user and, if valid, signed.
+* `settings` - *Optional*. Settings which are universally applicable to all `votes` being signed.  This is a json string.  The fields within the settings json string depend on the format of the vote record.  The wallet does not need to interpret this field and should pass it unchanged to the logic used to format the vote transaction.  However, to support multiple future vote transaction formats, the `settings` must contain at least:
```

Comment:
if the wallet doesn't validate the purpose in settings, can't a malicious actor abuse that to pass a different purpose than the one whitelisted by the wallet? If so, I'd consider removing the purpose from the settings and passing it separately, or at least specifying that the wallet should validate that field in the settings

Reply:
though looking at the js bidings lib to sign votes, I don't see the `purpose` field being used at all: https://github.com/input-output-hk/catalyst-core/blob/f5865be9a9d227e975689e32e6c03b5b34ff5d96/src/chain-wallet-libs/bindings/wallet-wasm-js/js-test/test/vote_cast.js#L10 - what's the benefit of passing `purpose` in the `settings` parameter then?

Reply:
The intent of including the purpose here is to allow wallets to validate. Wallets should be able to show the supplied purpose of a vote to user at signing time for them to validate.